cache/src/main/java/io/envoyproxy/controlplane/cache/SnapshotResources.java

```diff
@@ -37,5 +60,20 @@
   /**
    * Returns the version associated with this resources in this collection.
    */
-  public abstract String version();
+  public String version() {
+    return resourceVersionResolver().version();
+  }
+
+  /**
+   * Returns the version associated with this resources in this collection, where the key is the name of the resource.
```

Comment:
Maybe something like `Returns the snapshot version associated with the provided list of resources`?

This just returns a String, what does the "key" refer to here?